Mathlib/Computability/NFA.lean

```diff
@@ -112,6 +124,296 @@ theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
   rw [← toDFA_correct] at hx ⊢
   exact M.toDFA.pumping_lemma hx hlen
 
+/-! ### Regex-like operations
+
+Since NFAs are models for regular languages, they provide decision procedures for the empty set,
+singleton sets with members of at most length 1 as well as a notion of closedness under union,
+concatenation and application of the Kleene star.
+
+Cf. <https://en.wikipedia.org/wiki/Thompson%27s_construction> for the basic idea in a setting
+permitting ε-moves.
+-/
+instance : Zero (NFA α σ) :=
+  ⟨default⟩
```

Comment:
I think it's a bit strange to have this defined in terms of `default`, which is meant to be an arbitrary element of the type with no special meaning. I think it would be best to move this instance to the top of the file and then define the `Inhabited` instance as `⟨0⟩` and replace uses of `default` further down by `0`.

Reply:
Thank you for pointing that out! I didn't want to upend the sectioning of the file, so I instead moved the `Inhabited` instance down so it can use `⟨0⟩`. The uses of `default` further down refer to the states, which can, inherently, not depend on implementation details of `σ/σ'`.

A silly thought that occured to me to really drive the point of not relying on the particulars of `default` home was to just use `1`, which doesn't need `σ` to be `Inhabited` to be defined but has differing semantics depending on whether or not `σ` actually is.